File: backend/schemas/analytics.py
"""
Pydantic schemas for Placement Analytics API.
"""
from typing import Optional
from pydantic import BaseModel


class ScoreTrendPoint(BaseModel):
    date: str
    score: int
    label: str


class CompetencyBreakdown(BaseModel):
    hr: int
    technical: int
    dsa: int
    communication: int
    problem_solving: int


class SkillMasteredItem(BaseModel):
    skill: str
    frequency: int
    confidence: str


class SkillToImproveItem(BaseModel):
    skill: str
    frequency: int
    priority: str


class ActivityItem(BaseModel):
    id: str
    type: str
    title: str
    timestamp: str
    detail: Optional[str] = None


class RecommendationItem(BaseModel):
    category: str
    action: str
    impact: str


class AnalyticsSummaryResponse(BaseModel):
    overall_readiness_score: int
    readiness_category: str  # "Excellent", "Good", "Average", "Needs Improvement"
    motivational_summary: str

    # Resume Stats
    current_ats: Optional[int] = None
    highest_ats: Optional[int] = None
    average_ats: Optional[int] = None
    ats_trend: list[ScoreTrendPoint] = []

    # Job Match Stats
    latest_job_match: Optional[int] = None
    highest_job_match: Optional[int] = None
    average_job_match: Optional[int] = None
    job_match_trend: list[ScoreTrendPoint] = []

    # Interview Stats
    average_interview_score: Optional[int] = None
    best_interview_score: Optional[int] = None
    total_interviews: int = 0
    interview_trend: list[ScoreTrendPoint] = []
    competency_breakdown: CompetencyBreakdown

    # Learning Stats
    learning_progress_percentage: int = 0
    completed_weeks: int = 0
    remaining_weeks: int = 4

    # Skill Intelligence
    mastered_skills: list[SkillMasteredItem] = []
    skills_to_improve: list[SkillToImproveItem] = []

    # Activity Counts & Timeline
    total_resumes_uploaded: int = 0
    total_job_matches: int = 0
    total_interviews_taken: int = 0
    total_roadmaps_generated: int = 0
    recent_activities: list[ActivityItem] = []

    # Recommendations
    recommendations: list[RecommendationItem] = []

File: frontend/src/types/analytics.ts
export interface ScoreTrendPoint {
  date: string
  score: number
  label: string
}

export interface CompetencyBreakdown {
  hr: number
  technical: number
  dsa: number
  communication: number
  problem_solving: number
}

export interface SkillMasteredItem {
  skill: string
  frequency: number
  confidence: string
}

export interface SkillToImproveItem {
  skill: string
  frequency: number
  priority: string
}

export interface ActivityItem {
  id: string
  type: string
  title: string
  timestamp: string
  detail?: string | null
}

export interface RecommendationItem {
  category: string
  action: string
  impact: string
}

export interface AnalyticsSummary {
  overall_readiness_score: number
  readiness_category: string
  motivational_summary: string
  current_ats: number | null
  highest_ats: number | null
  average_ats: number | null
  ats_trend: ScoreTrendPoint[]
  latest_job_match: number | null
  highest_job_match: number | null
  average_job_match: number | null
  job_match_trend: ScoreTrendPoint[]
  average_interview_score: number | null
  best_interview_score: number | null
  total_interviews: number
  interview_trend: ScoreTrendPoint[]
  competency_breakdown: CompetencyBreakdown
  learning_progress_percentage: number
  completed_weeks: number
  remaining_weeks: number
  mastered_skills: SkillMasteredItem[]
  skills_to_improve: SkillToImproveItem[]
  total_resumes_uploaded: number
  total_job_matches: number
  total_interviews_taken: number
  total_roadmaps_generated: number
  recent_activities: ActivityItem[]
  recommendations: RecommendationItem[]
}
